packages/@blockprotocol/graph/src/custom-element.ts

```diff
@@ -5,19 +5,15 @@ import {
   EntityPropertiesObject,
   GraphBlockHandler,
 } from "./index.js";
+import { getRoots } from "./stdlib/subgraph/roots";
 
-export interface BlockElementBase<
-  BlockEntityProperties extends EntityPropertiesObject | null,
```

Comment:
I must say I didn't fully understand the purpose of all of these generic parameters (I think it was to let blocks tell TypeScript what the type of the Block Entity was). With my removal this isn't going to be possible, so perhaps we want to rework the `Subgraph` object to be able to even further constrain the typings of the `roots` with another generic parameter.

Probably by making `Entity` generic.

It's a little hard to figure this out prior to us knowing the shape of TS type generation from the type system.

Reply:
To note that the block templates won't compile now (because they are still passing a generic), but I suggest we just leave them because we're going to want to plug in a generic somehow.

Agree we can figure out what happens here later